### PR TITLE
util/selection: migrate test-infra to testify

### LINKED ISSUE #26415: migrate test-infra to testify for `util/selection` pkg

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #26415 

### Release note

- No release note

### CODE REVIEW

File: util/selection/selection_test.go [hunk 1]
```diff
@@ -58,37 +48,37 @@ func init() {
 	globalCase50 = randomTestCase(50)
 }
 
-func (s *testSuite) TestSelection(c *C) {
+func TestSelection(t *testing.T) {
```
Comment (mmyj) [inline util/selection/selection_test.go:51]: It seems that those test cases could be paralleled.

**Review by tisonkun — COMMENTED** [review]
Second @mmyj 's comment. All these tests can be run in parallel.

File: util/selection/selection_test.go (hunk 1 above)
Comment (feitian124) [reply util/selection/selection_test.go:51]: thanks, fixed and updated.

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by mmyj — APPROVED** [review]
lgtm

**Review by xhebox — APPROVED** [review]

**Pull request merged**